### ISSUE TITLE
Rule request: no overly large if statements

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [microsoft/tslint-microsoft-contrib#335]
This code has a lot of stuff in the `if` statement:
```typescript
function doSomething() {
    if (condition) {
        doSomethingA();
        doSomethingB();
        doSomethingC();
        doSomethingD();
        doSomethingE();
        doSomethingF();
        doSomethingG();
        return;
    }

    doSomethingZ();
}
```

It could be useful for a rule that requests `if` statements don't get unnecessarily large, since this unnecessary amount of nesting could be easily reduced by inverting the `if`.

```typescript
function doSomething() {
    if (!condition) {
        doSomethingZ();
        return;
    }

    doSomethingA();
    doSomethingB();
    doSomethingC();
    doSomethingD();
    doSomethingE();
    doSomethingF();
    doSomethingG();
}
```

The backing logic could be that if an `if` statement takes up more than (configuration: half, # of lines, combo of both) a function, it should be flagged?

**Comment by HamletDRC** [comment]
I feel like you'll end up with a lot of suppressed warnings in the code. Is length of the branch really how you decide what goes inside the if statement and what goes outside?

**Comment by JoshuaKGoldberg** [comment]
I can't think of a better way. Perhaps allowing a ratio and/or minimum statement count as a rule config?

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is being deprecated**; therefore, it and tslint-microsoft-contrib are no longer accepting pull requests for major new changes or features. See #4534. 😱

If you'd like to see this change implemented, you have two choices:
* **Recommended:** Check if this is available in [ESLint](https://eslint.org) + [typescript-eslint](https://typescript-eslint.io) ✅
* _Not Recommended: Fork TSLint or tslint-microsoft-contrib locally_ 🤷‍♂️

👋 It was a pleasure open sourcing with you!

_If you believe this message was posted here in error, please comment so we can re-open the issue!_
